### PR TITLE
Multithreaded georef (issues/#435)

### PR DESCRIPTION
**Pull request by pulquero**
Both best camera and best GCP multithreaded.
Seems to be working well.

### CODE REVIEW

File: modules/odm_georef/src/Georef.cpp [hunk 1]
```diff
@@ -1185,9 +1185,39 @@ void Georef::chooseBestGCPTriplet(size_t &gcp0, size_t &gcp1, size_t &gcp2)
 
 void Georef::chooseBestCameraTriplet(size_t &cam0, size_t &cam1, size_t &cam2)
 {
+    size_t numThreads = 2;
```
Comment (pierotofy) [inline modules/odm_georef/src/Georef.cpp]: Very nice! In the future in might be useful to change numThreads from 2 to a function that retrieves the number of available CPUs of the machine.

**Review by pierotofy — APPROVED** [review]

**Review by pierotofy — APPROVED** [review]

**Comment by pulquero** [comment]
If there is anything else that needs multithreading, let me know. I'll be willing to do it, now I got the hang of it. BTW, I've also noticed that in most places across all the modules, reserve() isn't being used to pre-allocate large vectors. Might be worth considering to save some heap thrashing.

**Comment by pierotofy** [comment]
@pulquero it would be interesting to run a profiling tool and detect the functions / modules that are taking the longest to compute. Those would be the ones to target for additional multithreading. Perhaps you might be interested in publishing the profiling results on the wiki.

**Comment by dakotabenjamin** [comment]
Thanks for this! I do not think `--time` is adequate for what @pierotofy is suggesting because it is too simplistic.

**Comment by xialang2012** [comment]
Ortho module is also slow too. So...

**Comment by pulquero** [comment]
I think the geotiff is to blame for that. It seems to take an extraordinary amount of time to generate the geotiff compared to the png.

**Comment by xialang2012** [comment]
The expensive part of the Ortho module is https://github.com/OpenDroneMap/OpenDroneMap/blob/master/modules/odm_orthophoto/src/OdmOrthoPhoto.cpp#L468.

**Comment by pulquero** [comment]
That looks like it can be multi-threaded, I'll have a go at it.

**Comment by dakotabenjamin** [comment]
That and https://github.com/OpenDroneMap/OpenDroneMap/blob/master/scripts/odm_orthophoto.py#L94 for large datasets. This is already multithreaded if you have an updated gdal. It always throws an unsupported warning for me though.

**Pull request merged**